api/src/room_store.py

```diff
@@ -50,6 +50,7 @@ def read_room_data(self, room_id: str) -> dict:
         if self._is_valid_path(full_path):
             with open(full_path, 'r') as f:
                 return json.loads(f.read())
+        raise KeyError(f"Unknown room id {room_id}")
```

Comment:
I think raising an exception here is a strange choice. Maybe we can return an optional instead?

Reply:
Great idea. I love how much you came up with it